app/models/player.rb

```diff
@@ -0,0 +1,31 @@
+class Player < ActiveRecord::Base
+  # has_many :hands
+  has_many :played_cards
+  has_many  :cards, :through => :played_cards
+
+
+  def draw_white_card
+    available_cards = Card.joins("LEFT JOIN played_cards ON played_cards.card_id = cards.id where played_cards.card_id is null and cards.color = 'white'")
+
+    card = available_cards.sample
+
+
+  end
+
+  def cards_in_hand
+    all_cards = self.played_cards.where(played: nil, ).map do |played_card|
+      Card.find(played_card.card_id)
```

Comment:
Doing finds in a look like this is super slow. If you have to do something manual like this then at least look up all the records in one query `cards = Card.where(id:[1,2,3])`